### PR TITLE
2hander update

### PR DESCRIPTION
**Pull request by Pebblesthe2nd**
## About The Pull Request

Picks up where I left off I guess. Was classifying melee weapons, making some sense of the arbitrary values and trying gimmicks, guess about halfway done. Starts on second half.

Tl:DR
* Gimmicks to 2handers polished. Fear the bat.
* Thermic lance and Chainsaw "fixed"
* 2h damage capped at 55, 1h at 45. Adjustments to fit the range done,
* D-claw spear and warhammer ripper taken down a notch
* Fixes/improves a bunch of my old 2h sprites, onmob and icons (fireaxe, spears, rocket hammers + legacy thermic and sledge gets shading)

Specifics
Thermic Lance redone.  Cost: manual on switch, and the turn on sound. Win: put it on your back, can hold in one hand (useless for fight), the bug where its copied is gone. Now has a little AP and bonus damage to metal doors and airlocks.

Chainsaw: Redone using 2h code. Loses some stuff as above, same win. Big bonus damage to wooden barricades.

Structure: 2h weapons capped at damage 55, 1h at 45, scaled from that so the top dog remains the same. Reason? With buffout/traits that boost damage just bumping up damage to make stuff more "elite" is a bad idea.

Gimmicks: Melee has tons of room for fun special effects, lets use some, and each class gets its own.
Fireaxes, bumperswords, boneaxes bonus to whacking doors
Sledgehammer, super sledges, rockethammers, small bonus to whacking all structures, barricades.
Bats are so boring, trying out the mini-knockback again, but only on the louisville for now. Stam damage fixed on bat, spiked bat and golf club.
Spears: Reach, low-ish damage, throwing power/embed to the scrap and bone spears so they got a fun little niche.

Two silly OP weapons are taken in the ear, the warhammer ripper sword being statwise a supersledge+ in one hand. Stays top 1h damage (45 now) and AP reduced to 0.1, so still stupid good (ripper got wound bonus to compensate no AP) but less broken.
Deathclaw spear at 43 damage and 0.3!!!! AP with reach is a remnant of the weird concept of having superpowered tribal weapons. Reduced to 38 and AP 0.1, second best spear legion lance bumped down from 40 to 34.  D-claw spear is still leagues better than the runner up, and still solid compared to other 2handers. Reach is super powerful, if they got the same or better damage as no reach 2handers its a nobrainer and bad design. Craftable weapons need to have SOME balance. Didn't make the Legion lance to be some superweapon either, just a flavorful and decent option for reach, suitable for more tribal type legionaires, not some mainstay replacing the iconic fireaxe.

Sprites: Just adds some shading and makes fireaxes, rocket hammers etc a little bit chunkier so it looks better when actually ingame, based on lessons learned. Probably no one else but me will notice, shading and highlights, small stuff. Fireaxe inhands being correctly flipped and shaded maybe somene might notice, or supersledge.

Shishkebab at 52 cause its sort of special case, subject to change.

## Why It's Good For The Game

Outlier broken damage stacks somewhat reduced. Some fun gimmicks added. Thermic Lance and Chainsaw less FUBAR.
Couple broken eyesores fixed and the value of reach more properly accounted for. It leaves melee in a better place.
Only risque thing is the knockback thing on the louisville, but its worth seeing how it pans out. Because its cool. 

## Pre-Merge Checklist
- [x] You tested this on a local server.
- [x] This code did not runtime during testing.
- [x] You documented all of your changes.

## Changelog
:cl:
balance: 2h damages roof 55, 1h 45, some gimmick abilites added to 2handers.
imageadd: 2h melee icons modest overhaul
code: thermic lance and chainsaw redone
/:cl:

### CODE REVIEW

**Comment by AdamElTablawy** [comment]
No thank you, Pebbles. This isn't very well balanced in regards to damage numbers (and overnerfs melee when it's already practically useless) and we really do not want to keep things in the forced twohanded category.

If you want to make a change, please make the weapons onehanded and then twohand-to-wield, like the baseball bat, etc, and leave their AP and damage values alone.

I have my own changes in mind for melee later.

File: code/game/objects/items/melee/f13twohanded.dm [hunk 1]
```diff
@@ -549,6 +595,11 @@ obj/item/twohanded/sledgehammer/supersledge/ComponentInitialize()
 	. = ..()
 	AddComponent(/datum/component/two_handed, force_unwielded = 15, force_wielded = 30, icon_wielded="[icon_prefix]2")
 
+/obj/item/twohanded/sledgehammer/shamanstaff/attack(mob/living/M, mob/living/user)
+	. = ..()
+	if(!istype(M))
+		return
+	M.apply_damage(30, STAMINA, null, 0)
```
Comment (AdamElTablawy) [inline code/game/objects/items/melee/f13twohanded.dm:602]: Thiiiiiiiis will just directly do 30 stam damage regardless of armor per hit.

**Comment by Pebblesthe2nd** [comment]
> No thank you, Pebbles. This isn't very well balanced in regards to damage numbers (and overnerfs melee when it's already practically useless) and we really do not want to keep things in the forced twohanded category.
> 
> If you want to make a change, please make the weapons onehanded and then twohand-to-wield, like the baseball bat, etc, and leave their AP and damage values alone.
> 
> I have my own changes in mind for melee later.

Maybe spend a minute reading and understanding the PR before commenting is in order. 
You haven't even understood that the Thermic Lance and Chainsaw currently are forced 2h, and my PR changes that.
Claiming it nerfs melee as a whole is also garbage, raw damage is only lowered for a few of the very top, others like fireaxes get a slight boost, and the bonus damage to structures etc bring new utility to melee. 
The stun damage has been there all this time for bats, and somehow it hasnt broken the game, the values might need tweaking for some but pretending like its turbo broken is a load of bs.

You've had half a year to do your own changes, and I'm still waiting on the armor refactor that was promised back in like may, so yeah, no, don't need your hot take.

**Comment by AdamElTablawy** [comment]
> > No thank you, Pebbles. This isn't very well balanced in regards to damage numbers (and overnerfs melee when it's already practically useless) and we really do not want to keep things in the forced twohanded category.
> > If you want to make a change, please make the weapons onehanded and then twohand-to-wield, like the baseball bat, etc, and leave their AP and damage values alone.
> > I have my own changes in mind for melee later.
> 
> Maybe spend a minute reading and understanding the PR before commenting is in order. You haven't even understood that the Thermic Lance and Chainsaw currently are forced 2h, and my PR changes that. Claiming it nerfs melee as a whole is also garbage, raw damage is only lowered for a few of the very top, others like fireaxes get a slight boost, and the bonus damage to structures etc bring new utility to melee. The stun damage has been there all this time for bats, and somehow it hasnt broken the game, the values might need tweaking for some but pretending like its turbo broken is a load of bs.
> 
> You've had half a year to do your own changes, and I'm still waiting on the armor refactor that was promised back in like may, so yeah, no, don't need your hot take.

Regardless of if you need my ‘hot take’ or not, what I was saying is that the only part of this PR that’s needed is the onehand-twohand part, which is what I was saying (and communicated badly) above.

The rest is unnecessary, and you can remove those at lesiure or I can do it for you.

**Review by OneOneThreeEight — CHANGES_REQUESTED** [review]
Yeah hey,

The nerfs aren't justifiable in their current state. I love the QOL fixes to make them, you know, actually work in terms of inventory, but nerfing their damage and armor pen capability is a no-go here. Have them adjusted to prior damage values. Melee weapons are in practicality, very terrible weapons as-is, dragging them down further at current state can't work right now for game balance's sake.

**Comment by Pebblesthe2nd** [comment]
> Yeah hey,
> 
> The nerfs aren't justifiable in their current state. I love the QOL fixes to make them, you know, actually work in terms of inventory, but nerfing their damage and armor pen capability is a no-go here. Have them adjusted to prior damage values. Melee weapons are in practicality, very terrible weapons as-is, dragging them down further at current state can't work right now for game balance's sake.

Mmm, fair enough, though I think I have to load the parapost.exe here to explain the whole concept behind the PR, hold on:

It should be pointed out that making the spread of damage less in melee has to do with internal balance and logic for melee. This effort was aimed at internal melee balance more than towards buffing damages to compete with the gun/armor spiral. Compound changes from guns and traits makes balancing by just bumping up numbers for a select few items iffy. 
For balancing it in general pvp you have gun damage, the effect of armor as obvious ones, sprint being deleted impacts things quite a bit too, as does the effect of drugs and traits wounds etc., all which needs gradual adjustments. 

Theres more than one issue, you have the fact that melee revolves around a small handful of the most extreme outliers. I believe multiplayer games are more fun and balanced when upgrades and the like are subtler and you still can compete with second hand stuff, which you cant if the step between stuff is multiples. Then you gradually shift the ranges and adjust other factors.

For internal melee balance, you have a 1h weapon thats essentially the supersledge but faster, and a craftable spear thats leagues and bounds better than any other in its category. This is not fine, machetes and bumper swords and whatnot being almost useless is the result of this way of thinking that smacking huge values on item X solves stuff, when it leaves items F, U and C as utter garbage.

The only things I have seen done melee wise at a glance for a long time are:
* Deletion of sprint: Impacts the ability to get into melee in a fight obviously.
* The buggy firearm/melee thing was fixed, which was great.
* Proliferation of the most extreme weapons: The military ripper was made half as a meme because the maker likes warhammer stuff and wanted BoS to have melee power, and thats honestly fine and all, but now its not just a unique thing for one leader class, another BoS class is getting it, as well as the centurion. The Fallout ripper becomes almost a joke, the warhammer sword is the new normal. 

Slapping in the currently top damage dealers for say, Legion, isn't an improvement either in my eyes, veterans getting spears instead of fireaxes is just not as cool, nor is decanus running around with a supersledge, or the cent copying the bos sword instead of using his iconic thermic lance. Its powercreep, not that it makes melee good, but it does highlight the issue with the handful of outliers being leagues better than the second best stuff. 

Just a few examples: Fixing the thermic should have been done instead of throwing a cool thing away. The Military ripper thing could have been JUST given some AP, it didn't need to be supersledge level (50 dam + ap 0.15 and 1h speed is pretty equal to 60 dam 0 ap 2h speed if not better). I think having the iconic stuff being the top dog is the way to go, so super sledge should clearly be better than the warhammer ripper, but so should the Fallout one! 

Same with Legion lance, I didnt make the Lance to replace the fireaxe, I made it in another era when the meta was completely different, its a whole story of its own but it doesn't matter. It, like the military ripper, is homebrew stuff, and while I like it, it should NOT be superior to the classics, just a sidegrade. 

Making homebrew stuff like these and deathclaw spear the go to thing is pointless, that niche could just as well be filled by the original item, just tweaked, and I think its cooler to keep the oldschool things viable. Homebrews should NEVER exceed the established stuff in power in my view, thats how power creep spirals out of control.

Point of this PR isnt really exact damage values, those are always adjustable and up for debate, the key is in spread, and the tinkering with abilites, such as making fireaxes great for chopping doors, sledges for barricades, chainsaws superlative for wooden barricades etc. Adding utility to stuff is in my opinion a fun way to make things useful outside pure pvp as well, thats why I made the rocket hammer and gave it the digging ability just as an example, cause its fun. 
My idea was to use AP from the baseline to find the right spot to avoid compound issues until the root causes, gun and armors are adjusted, but not just for the outliers, keeping it more consistent to make the fun mid stuff useful. Not everything can be done in one go though.


Either way, this is more than enough words on the matter, this is a place as good as any to explain the logic of the thing for posterity though, not that anyone will do much other than cherrypick a line and use it to trashtalk me in half a year. Have at it.

For adjusting the PR, I'd already decided to take a break. so hadn't intended to do any more work. I'll consider it, but I'll be honest, my motivation right now is super low, the lack of good faith has just been exhausting, and a reminder why I was on a long break before.

I'll see whats what in a week or so, for now, best of luck.

**Comment by OneOneThreeEight** [comment]
The maintainer team appreciate your attempts at contribution nonetheless. If you feel the need to take a break that's completely fine.

**Pull request closed without merging**